### PR TITLE
Update generated docs after recent metadata format changes affecting …

### LINKED ISSUE #1221: Remove nitclk properties that need repeated capabilities
Two properties in NI-TClk need repeated capabilties.
We don't support that in `nitclk` so we will remove it. We don't think the use cases justify it.

The properties are:
* `NITCLK_ATTR_SCRIPT_TRIGGER_MASTER_SESSION`
* `NITCLK_ATTR_SEQUENCER_FLAG_MASTER_SESSION`

If a use case emerges where these are needed, we can reconsider solving the limitation.

### LINKED ISSUE #1607: Update metadata of drivers to add the rep cap tip back to applicable properties
In #1599 the metadata key for rep cap tip was updated. Which means the tip is missing for all the applicable properties in all modules except nidcpower.

More info:
_Originally posted by @luisgomes252 in https://github.com/ni/nimi-python/pull/1599#discussion_r632087340_

### PR DESCRIPTION
**Pull request by luisgomes252**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [ ] ~~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~
- [ ] ~~I've added tests applicable for this pull request~~

### What does this Pull Request accomplish?

Update generated docs after recent metadata format changes affecting the channel_based, supported_rep_caps and repeated_capability_type tags

### List issues fixed by this Pull Request below, if any.

* Fix #1607
* Fix #1221

### What testing has been done?

Performed internal builds and inspected generated code, while verifying that all automated tests pass.

### CODE REVIEW

**Comment by sbethur** [comment]
ok to test

**Review by sbethur — CHANGES_REQUESTED** [review]

File: src/nidigital/metadata/attributes.py [hunk 1]
```diff
@@ -985,45 +1014,48 @@
     1150072: {
         'access': 'read-write',
         'attribute_class': 'AttributeViReal64TimeDeltaSeconds',
-        'channel_based': False,
         'documentation': {
             'description': 'Specifies a timing delay, measured in seconds, and applies the delay to the digital pattern instrument in addition to TDR and calibration adjustments. If the NIDIGITAL_ATTR_TIMING_ABSOLUTE_DELAY_ENABLED attribute is set to VI_TRUE, this value is the intermodule skew measured by NI-TClk. You can modify this value to override the timing delay and align the I/O timing of this instrument with another instrument that shares the same reference clock. If the NIDIGITAL_ATTR_TIMING_ABSOLUTE_DELAY_ENABLED attribute is VI_FALSE, this attribute will return 0.0. Changing the NIDIGITAL_ATTR_TIMING_ABSOLUTE_DELAY_ENABLED attribute from VI_FALSE to VI_TRUE will set the NIDIGITAL_ATTR_TIMING_ABSOLUTE_DELAY value back to your previously set value.\n'
         },
         'name': 'TIMING_ABSOLUTE_DELAY',
-        'repeated_capability_type': 'instruments',
```
[the remark this answers is not in the document]
Comment (luisgomes252) [reply src/nidigital/metadata/attributes.py:993]: That's interesting. I hadn't taken the approach of doing a direct search & replace on this file. Instead, I simply regenerated the file from the metadata, on the assumption that the metadata was correct, with regards to the repcaps. (I did remove the other non-repcap related changes that were generated).

The metadata for TIMING_ABSOLUTE_DELAY didn't list a repcap. I can add an 'instruments' repcap to it, if it belongs there. I then looked for other discrepancies and noticed that TIMING_ABSOLUTE_DELAY_ENABLED does have an 'instruments' repcap in the metadata, even though it wasn't listed as such in this repo. Can I assume that the metadata is correct for that property and leave it in?

File: src/nidigital/metadata/attributes.py [hunk 2]
```diff
@@ -232,11 +213,13 @@
         },
         'enum': 'SelectedFunction',
         'name': 'SELECTED_FUNCTION',
+        'supported_rep_caps': [
+            'channels'
```
[the remark this answers is not in the document]
Comment (luisgomes252) [reply src/nidigital/metadata/attributes.py:217]: Ok, will do.

File: src/nidigital/metadata/attributes.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (luisgomes252) [reply src/nidigital/metadata/attributes.py:993]: Actually, I was wrong about TIMING_ABSOLUTE_DELAY_ENABLED -- the metadata did not have it with a repcap. So the only discrepancy was TIMING_ABSOLUTE_DELAY.

[the remark this answers is not in the document]
Comment (sbethur) [reply src/nidigital/metadata/attributes.py:993]: The GitHub copy is the "source of truth" for nidigital metadata.

TIMING_ABSOLUTE_DELAY supports instruments rep cap.
TIMING_ABSOLUTE_DELAY_ENABLED does not.

**Comment by sbethur** [comment]
test this please

**Review by sbethur — APPROVED** [review]

File: docs/nitclk/class.rst [hunk 3]
```diff
@@ -183,7 +164,7 @@ finish_sync_pulse_sender_synchronize
             
 
 
-        :type sessions: list of (nimi-python Session class or nitclk.SessionReference)
+        :type sessions: list of (Driver Session or nitclk.SessionReference)
```
Comment (sbethur) [inline docs/nitclk/class.rst:167]: I think we should keep the old version - `nimi-python Session class`. @marcoskirsch, can you please confirm.

Comment (luisgomes252) [reply docs/nitclk/class.rst:167]: This was a case in which the config file had duplicate entries, so I removed one of them (based on consistency with the rest of the functions in the API, but I'm not completely sure). Note the functions.py diff:

image

Comment (marcoskirsch) [reply docs/nitclk/class.rst:167]: >I think we should keep the old version - nimi-python Session class. @marcoskirsch, can you please confirm.

The average person calling `pip install niscope` has no good reason to be familiar with the term `nimi-python`. 99% of the clients using our Python APIs are going to get them from PyPI. I actually want to open an Issue about the fact that https://pypi.org/project/niscope/ talks about nimi-python and mentions things beyond `niscope`.

All this said, type_in_documentation can be `nifgen.Session`, `niscope.Session`, `nidigital.Session`, `niscope.Session` _or_ `nitclk.SessionReference`. Not sure what the best way to refer to these generically is.

Comment (sbethur) [reply docs/nitclk/class.rst:167]: That makes sense.

We have used "Session instance" [elsewhere](https://nimi-python.readthedocs.io/en/master/search.html?q=%22Session+instance%22&check_keywords=yes&area=default#) in documentation. But here we also need to differentiate between nitclk and other modules. How about:
```
'list of instrument Session or nitclk SessionReference instances'
```

Comment (marcoskirsch) [reply docs/nitclk/class.rst:167]: yes

Comment (marcoskirsch) [reply docs/nitclk/class.rst:167]: minor tweak

'list of instrument-specific Session or nitclk.SessionReference instances'

Comment (luisgomes252) [reply docs/nitclk/class.rst]: Ok, pushed new changes based on [updated](https://ni.visualstudio.com/DevCentral/_git/ni-central/pullrequest/172689) metadata.

**Review by marcoskirsch — APPROVED** [review]

**Pull request merged**